src/Entity/SmsMessage.php

```diff
@@ -587,12 +594,92 @@ public static function convertFromSmsMessage(StdSmsMessageInterface $sms_message
    */
   public static function postDelete(EntityStorageInterface $storage, array $entities) {
     parent::postDelete($storage, $entities);
+    $results = [];
+    $reports = [];
+    /** @var \Drupal\sms\Entity\SmsMessageInterface $sms_message */
     foreach ($entities as $sms_message) {
-      $sms_message->getResult() && $sms_message->getResult()->delete();
+      // Since the $sms_message can have both in-memory and stored objects, only
+      // need to delete actual stored entities.
+      if (($result = $sms_message->getResult()) && $result instanceof EntityInterface) {
+        $results[] = $result;
+      }
       foreach ($sms_message->getReports() as $report) {
-        $report->delete();
+        if ($report instanceof EntityInterface) {
+          $reports[] = $report;
+        }
       }
     }
+    \Drupal::entityTypeManager()->getStorage('sms_result')->delete($results);
+    \Drupal::entityTypeManager()->getStorage('sms_report')->delete($reports);
```

Comment:
Yeah. I've caught myself making this change several times also - until I remember it's a static method. :)